lib/database/leaderboard.php

```diff
@@ -449,7 +467,8 @@ function GetLeaderboardData($lbID, $user, $numToFetch, $offset, $friendsOnly)
             while ($db_entry = mysqli_fetch_assoc($dbResult)) {
                 $db_entry['DateSubmitted'] = strtotime($db_entry['DateSubmitted']);
                 settype($db_entry['Score'], 'integer');
-                $db_entry['Rank'] = (int) $db_entry['UserRank'];
+                $db_entry['Rank'] = (int) $db_entry['Rank'];
+                $db_entry['UserIndex'] = (int) $db_entry['UserIndex'];
```

Comment:
`UserIndex` is misleading. this value is just the offset of the entry in the query to be fed back into the API to fetch additional neighboring records. suggest either `Offset` (to match `o` parameter) or `Index`.

Reply:
Both Offset and Index are MySql keywords so I didn't want to use them for that reason (need to change Rank as well) at least in the query but I think I can update the response names to be those and just change the query names. I'll look into this

Reply:
Rank definitely needs to not be renamed in the outgoing message, or it will break the existing API usages. You can see it was doing that previously:
```
                $db_entry['Rank'] = (int) $db_entry['UserRank'];
```
So it should be easy enough to do for `Offset` or `Index` as well.

This also explains why both were being sent to the client and were identical. The "rename" wasn't removing the old name.